### PR TITLE
Add the Practitioner selection screen on the In-app reporting workflow

### LINKED ISSUE #2472: [In-app Reporting] Add the Practitioner selection screen on the In-app reporting workflow.
**Describe the feature request.**
- Supervisors need the ability to filter the in-app reports based on the different practitioners they supervise. This means we need to provide an option for them to select the practitioner while on the reports page. 
- This screen should however be hidden based on a config for patient-centric applications
- Below is the screenshot of the desired reporting workflow. 

Screenshot 2023-06-14 at 21 48 46

**Acceptance criteria**
- [x] Show a list of practitioners on the first screen of in-app reports. 
    - [x] This list should be configurable like the other registers on the app. 
    - [ ] The single practitioner in-app reporting should be enabled if this page is shown
- [ ] Add the ability to hide to the Practitioner selection screen. 
    - [ ] The single practitioner in-app reporting should be disabled if this page is hidden  

**Implementation plan (For Engineers)**

- [x] Add register configs for displaying practitioners
- [x] Update `navigation_config` to launch the practitioner register
- [x] Add action to launch report screen when practitioner row is clicked using `LAUNCH_REPORT` workflow
- [x] Update logic for generating reports to add pass `practitionerId` to the `evaluatePopulationMeasure` function
- [x] Refactor navigation logic to open the `Module selection` screen before the `Report date selection` screen

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2472 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt [hunk 1]
```diff
@@ -104,7 +105,15 @@ fun List<ActionConfig>.handleClickEvent(
         }
       }
       ApplicationWorkflow.LAUNCH_REPORT -> {
-        val args = bundleOf(Pair(NavigationArg.REPORT_ID, actionConfig.id))
+        val interpolatedParams = interpolateActionParamsValue(actionConfig, resourceData)
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt:108]: Can this interpolation be done before the `when` clause so all branches can benefit from it?

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/quest/src/test/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensionsTest.kt [hunk 2]
```diff
@@ -183,14 +183,23 @@ class ConfigExtensionsTest : RobolectricTest() {
...
       )
     listOf(clickAction).handleClickEvent(navController = navController, resourceData = resourceData)
     val slotInt = slot<Int>()
     val slotBundle = slot<Bundle>()
     verify { navController.navigate(capture(slotInt), capture(slotBundle)) }
     Assert.assertEquals(MainNavigationScreen.Reports.route, slotInt.captured)
-    Assert.assertEquals(1, slotBundle.captured.size())
+    Assert.assertEquals(2, slotBundle.captured.size())
     Assert.assertEquals("reportId", slotBundle.captured.getString(NavigationArg.REPORT_ID))
+    Assert.assertEquals("practitioner_id", slotBundle.captured.getString(NavigationArg.RESOURCE_ID))
```
Comment (pld) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensionsTest.kt:202]: this assert is returning null

Comment (Rkareko) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensionsTest.kt:202]: This has been fixed

**Review by ellykits — DISMISSED** [review]
:+1: LGTM

**Review by pld — APPROVED** [review]

**Pull request merged**